### DISCUSSION TITLE
issue: Validation issue after unmounting fields

### DISCUSSION THREAD

**Discussion by KaranLala** [react-hook-form/react-hook-form#11897] [Q&A]
### Version Number

7.51.3

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/react-hook-form-7-51-3-validation-bug-z7yz34?file=%2Fsrc%2Fvalidation-not-working.js

### Steps to reproduce

1. Go to 'https://codesandbox.io/p/sandbox/react-hook-form-7-51-3-validation-bug-z7yz34?file=%2Fsrc%2Fvalidation-not-working.js'
2. Click on submit for the validation working from
3. Notice the correct errors show up, one for text input and one for textarea
4. Click on submit for the Validation For First Element Only Working Form
5. Notice the error only show up first text input field

### Expected behaviour

Not sure if it is expected or not, but its an easy trap to fall into. IMO isSubmitting should be true after validation. If this is as expected, a note about this in docs, or a new state that can be used to safely unmount the form during asynchronous operations would be helpful. 
Using isValid && isSubmitting together seems to do the trick. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
> IMO isSubmitting should be true after validation. 

not clear on what's reported here.

**Comment by KaranLala** [comment]
did you see the codesandbox and see that validation is not working as expected in the second form? i.e. validation is only being applied to the first input in the second form, when it should validate all fields. 

This happens when the form is unmounted when isSubmitting is set to true. If this is expected then nothing needs to be fixed.

**Comment by bluebill1049** [comment]
> did you see the codesandbox and see that validation is not working as expected in the second form? i.e. validation is only being applied to the first input in the second form, when it should validate all fields.

it wouldn't, rhf focus on single field validation at a time by design. if you need multiple use `trigger`

**Comment by KaranLala** [comment]
It's okay if RHF only validates one field at a time, however in this case, it only ever validates the first field. If you enter a value in the first field in the second form, it does not actually validate anything. You can see the image below. 

![react-hook-form-2](https://github.com/react-hook-form/react-hook-form/assets/5331717/b0da8553-db7c-4ae6-b9f5-3216db0c7fd0)

**Comment by erashu212** [comment]
Hi @KaranLala ,

In your `InvalidForm` you have conditional rendering
```jsx
if (isSubmitting) return <div>Loading...</div>;
```
as soon as you click the submit button this sets as true which unmount other elements but validations are still in process for other form field.

when the `RHF` goes for other field like `textarea` in your case, found it as unmount that is why validation does not appear.

You can try this way if you want all error message to be appeared

```jsx
 let {
    control,
    handleSubmit,
    formState: { errors, isSubmitting },
  } = useForm({ criteriaMode: "all" });
```

**Reply by KaranLala** [reply]
Thank you for your response. I've updated the code sandbox to include the criteriaMode option but it still doesn't seem to work. https://codesandbox.io/p/sandbox/react-hook-form-7-51-3-validation-bug-z7yz34?file=%2Fsrc%2Fvalidation-not-working.js%3A10%2C15

**Comment by zhangwc777** [comment]
Hello, I had the same problem, but I'm using a virtual list. I went down to 7.0.0 and it pretty much fixed the problem. This is a different link. What is the final solution to this kind of problem?   https://github.com/react-hook-form/react-hook-form/issues/8740

**Comment by zhangwc777** [comment]
What I'm going to do is validate all the fields in the useForm. Instead of the viewport, and keep the error message. Because I need to properly show validation failures when scrolling.
